fizzbuzz/src/main/java/fizz.buzz/number/factory/AbstractNumberFactory.java

```diff
@@ -0,0 +1,37 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
+
+import java.lang.reflect.InvocationTargetException;
+
+public abstract class AbstractNumberFactory implements NumberProcessor, NumberFactory {
```

Comment:
I'm not sure about the naming of the AbstractNumberFactory.
RegularNumberFactory is not extending it, but just based on the name I would expect it to be based on the abstract for number factory.

Also in the processing chain I would expect to have RegularNumberFactory implement setSuccessor. Just because the naming schema is the same.